### PR TITLE
feat(oui-datagrid): add expandable row detail

### PR DESCRIPTION
**Pull request by JeremyDec**
## Datagrid - Expandable row detail
31d0b38 - add expandable row detail

### :rocket: Feature

Add the possibility to add detail for each row of the dataset.
The row detail will make the row expandable to show details of it.

This feature is based on the work made on this PR : https://github.com/ovh-ux/ovh-ui-kit/pull/403

### CODE REVIEW

File: packages/components/datagrid/src/js/datagrid.html [hunk 1]
```diff
@@ -44,14 +48,25 @@
...
                             <oui-checkbox model="$ctrl.selectedRows[$index]"
                                             on-change="$ctrl.toggleRowSelection($index, modelValue)">
                             </oui-checkbox>
                         </td>
+                        <td class="oui-datagrid__cell oui-datagrid__cell-short"
+                            ng-if="$ctrl.expandableRows">
+                            <button type="button"
+                                class="oui-datagrid__expand-button"
+                                ng-if="$ctrl.expandableRows">
+                                <span ng-show="!$ctrl.isRowExpanded($index)" class="oui-icon oui-icon-chevron-down" aria-hidden="true"></span>
```
Comment (FredericEspiau) [inline packages/components/datagrid/src/js/datagrid.html]:
```suggestion
                                <span ng-if="!$ctrl.isRowExpanded($index)" class="oui-icon oui-icon-chevron-down" aria-hidden="true"></span>
```

File: packages/components/datagrid/src/js/datagrid.html [hunk 2]
```diff
@@ -44,14 +48,25 @@
...
+                                <span ng-show="$ctrl.isRowExpanded($index)" class="oui-icon oui-icon-chevron-up" aria-hidden="true"></span>
```
Comment (FredericEspiau) [inline packages/components/datagrid/src/js/datagrid.html]:
```suggestion
                                <span ng-if="$ctrl.isRowExpanded($index)" class="oui-icon oui-icon-chevron-up" aria-hidden="true"></span>
```

**Comment by ovh-cds** [comment]
CDS Report yarn-build#1407.0 ✘  
* build &amp; test
  * build ✘  
  * test ✘

File: packages/components/datagrid/src/js/datagrid.html [hunk 3]
```diff
@@ -66,24 +85,32 @@
...
                             <oui-datagrid-cell ng-if="row && $ctrl.hasActionMenu"
                                 row="row"
                                 column="$ctrl.actionColumn"
                                 index="rowIndex">
                             </oui-datagrid-cell>
                         </td>
                     </tr>
+                    <tr class="oui-datagrid__row"
+                        ng-repeat-end>
+                        <td class="oui-datagrid__row-detail-container"
```
Comment (AxelPeter) [inline packages/components/datagrid/src/js/datagrid.html]: `oui-datagrid__row-detail-container`, `oui-datagrid__row-detail`.
It seems like a sort of Block-Element-Element than just Block-Element.

I think we should create another Block level, like `oui-datagrid-row-detail`. And then have Elements like `oui-datagrid-row-detail__container`.

File: packages/components/datagrid/src/js/datagrid.html [hunk 4]
```diff
@@ -66,24 +85,32 @@
...
+                        <td class="oui-datagrid__cell oui-datagrid__cell-sticky"
+                            ng-if="$ctrl.hasActionMenu || $ctrl.customizable">
                             <oui-datagrid-cell ng-if="row && $ctrl.hasActionMenu"
                                 row="row"
                                 column="$ctrl.actionColumn"
                                 index="rowIndex">
                             </oui-datagrid-cell>
                         </td>
                     </tr>
+                    <tr class="oui-datagrid__row"
```
[the remark this answers is not in the document]
Comment (JeremyDec) [reply packages/components/datagrid/src/js/datagrid.html]: @AxelPeter it is done this way to make row background-color alternating correctly, here is more explanations :

With the ng-if on the `<tr>` element

If you have one row with the row-detail expanded, you'll have a row compiled with two `<tr>` elements. Others row will have only one `<tr>` element.
So, the nth-child(odd/even), will count this way : 

**Rows without any row detail displayed** :
All rows have a background color alternating correctly : 1st row bc = white, 2nd row bc = grey, 3rd row bc = white etc... all good

**First row with detail display** 
First row, => the row `background-color : white` 
               => the row-detail `background-color : grey`.

Next row : background-color : white.

Ok for the class name

**Comment by ovh-cds** [comment]
CDS Report yarn-build#1407.1 ✘  
* build &amp; test
  * build ✘  
  * test ✘

**Comment by ovh-cds** [comment]
CDS Report yarn-build#1412.0 ✘  
* build &amp; test
  * build ✔ 
  * test ✘

File: packages/components/datagrid/src/js/datagrid.controller.js [hunk 5]
```diff
@@ -4,7 +4,7 @@ import { hasProperty } from './util';
 
 import template from './datagrid.html';
 
-const cssSorted = 'oui-datagrid__header_sorted';
+const cssSorted = 'oui-datagrid__cell_sorted';
```
Comment (AxelPeter) [inline packages/components/datagrid/src/js/datagrid.controller.js]: Is there a reason that this classname has changed ?
The style is still applied to `oui-datagrid__header_sorted`.

Comment (JeremyDec) [reply packages/components/datagrid/src/js/datagrid.controller.js]: No, just a mistake

**Pull request merged**